R/gwdt.R

```diff
@@ -0,0 +1,35 @@
+#' wrap_gwdt
+#'
+#' Perform the grey-weighted distance transform (GWDT) on the DEM.
+#' 
+#' @param DEM (terra::SpatRaster) Digital elevation model
+#'
+#' @import terra
+#'
+#' @return A terra::SpatRaster representing the GWDT distances for each grid cell.
+#' 
+#' @export
+
+gwdt <- function(DEM){
+  DEMf <- fillsinks(DEM)
+  FLATS <- identifyflats(DEM)
```

Comment:
The flat regions need to be identified in the filled DEM, not the original one. This should be

```R
FLATS <- identifyflats(DEMf)
```